File: src/test/java/kitchenpos/helper/OrderTableHelper.java
package kitchenpos.helper;


import kitchenpos.domain.OrderTable;

import java.util.UUID;

public final class OrderTableHelper {

    public static final String DEFAULT_NAME = "테스트 기본 테이블명";

    private OrderTableHelper() {
    }

    public static OrderTable create() {
        return create(DEFAULT_NAME);
    }

    public static OrderTable create(int numberOfGuests) {
        OrderTable orderTable = new OrderTable();
        orderTable.setNumberOfGuests(numberOfGuests);
        return orderTable;
    }

    public static OrderTable create(String name) {
        OrderTable orderTable = new OrderTable();
        orderTable.setId(UUID.randomUUID());
        orderTable.setName(name);
        return orderTable;
    }

}

File: src/test/kotlin/kitchenpos/ordertable/OrderTableHelper.kt
package kitchenpos.ordertable

import io.restassured.RestAssured
import io.restassured.response.Response
import kitchenpos.domain.OrderTable
import java.util.*

class OrderTableHelper {
    companion object {
        fun 가게테이블이름으로_가게테이블_조회(name: String): OrderTable {
            return RestAssured
                .`when`().get("/api/order-tables")
                .then().extract().jsonPath().getList("", OrderTable::class.java)
                .find { it.name == name }!!
        }

        fun 가게테이블_생성(orderTable: OrderTable): Response {
            return RestAssured
                .given().body(orderTable).contentType("application/json")
                .`when`().post("/api/order-tables")
        }

        fun 가게테이블_생성_ID추출(orderTable: OrderTable): UUID {
            return 가게테이블_생성(orderTable).then().log().all().extract().jsonPath().getUUID("id")
        }

        fun 가게테이블_점유(orderTableId: UUID): Response {
            return RestAssured
                .put("/api/order-tables/$orderTableId/sit")
        }
    }
}
